### DISCUSSION TITLE
DS18B20 Temp Sensors

### DISCUSSION THREAD

**Discussion by dcan999** [iNavFlight/inav#9732] [Q&A]
I've been trying to get the DS18B20 working for awhile now with no luck. INAV 7.0/7.1 on a Matek F405-WMNm(also tried a F405-Wing V2 and a H743-Wlite with no luck)

I have tried the basic wiring config of:

DS2482 pins 

1-VCC ( to 5V on FC and also to VCC pin on DS18B20)
2-DQ (to center pin on DS18B20)
3- Ground (to FC ground and Ground pin on DS18B20)
4 -SCL (to SCL on FC)
5-SDA (to SDA on FC)
6 - NA (for parasitic use only)
7- AD1 (to Ground)
8-AD0 (to Ground). 

I have tried with and without a 4.7k pullup between DQ and VCC, and I've tried with and without 4.7k pullups on SCL/SDA. 

I've also tried all of the different configurations of AD1 and ADO to VCC and Ground. 

The sensor is never found or listed in CLI after a temp_sensor command, and I can't just manually enter the address like an LM75 since it's that long 64bit one stored on the sensor. 

I've tried 100, 200, and 400 khz for the I2C speeds as well, nothing works. 

I hooked up the same config above to an arduino and it works, so what's going on?

**Comment by usbdem** [comment]
I found the same problem. After looking at the source code I found the solution, but it requires rebuilding the firmware
In my case, in the firmware for my board in the file `src/main/target/[YOUR_BOARD_NAME]/target.h`, I added the line:
`#define TEMPERATURE_I2C_BUS BUS_I2C1` 
and everything worked!

**Reply by ailail2353** [reply]
Is there a crack in the firmware rebuild process?

**Comment by dcan999** [comment]
Thank you for that information. 

Hopefully this will be added in INAV 8, I am not good at building targets.
